lib/call-cabal-project-to-nix.nix

```diff
@@ -239,12 +238,13 @@ let
       #   on the target system would use, so that the derivation is unaffected
       #   and, say, a linux release build job can identify the derivation
       #   as built by a darwin builder, and fetch it from a cache
-      sourceReposEval = builtins.map (fetchRepo pkgs.evalPackages.fetchgit) sourceRepoData;
-      sourceReposBuild = builtins.map (x: (fetchRepo pkgs.fetchgit x).fetched) sourceRepoData;
+      sourceReposEval = builtins.map (fetchPackageRepo pkgs.evalPackages.fetchgit) sourceRepoPackageResult.sourceRepos;
+      sourceReposBuild = builtins.map (x: (fetchPackageRepo pkgs.fetchgit x).fetched) sourceRepoPackageResult.sourceRepos;
     in {
       sourceRepos = sourceReposBuild;
+      inherit (repoResult) tarballs extra-hackages;
       makeFixedProjectFile = ''
-        cp -f ${otherText} ./cabal.project
+        cp -f ${pkgs.evalPackages.writeText "cabal.project" repoResult.otherText} ./cabal.project
```

Comment:
what about the `otherText` from `souceRepoPackageResult`?

Reply:
It is actually passed as the input to `parseRepositories`